src/Microsoft.OData.Edm/ExtensionMethods/EdmTypeSemantics.cs

```diff
@@ -1216,18 +1221,33 @@ public static EdmPrimitiveTypeKind PrimitiveKind(this IEdmTypeReference type)
         /// <param name="potentialBaseType">The potential base type of the type being tested.</param>
         /// <returns>True if and only if the type inherits from the potential base type.</returns>
         public static bool InheritsFrom(this IEdmStructuredType type, IEdmStructuredType potentialBaseType)
+        {
+            HashSet<string> baseTypes;
+            string fullName = type.FullTypeName();
+
+            if (!baseTypeCache.TryGetValue(fullName, out baseTypes))
+            {
+                baseTypes = new HashSet<string>();
+
+                GetBaseTypes(type, baseTypes, fullName);
+            }

```

Comment:
actually, I would add this logic to GetBaseTypes as well; GetBaseTypes should check the cache to see if it has it and if not, create it and add it to the cache.  InheritsFrom should just check to see if GetBaseTypes(type).Contains(potentialBaseType)